src/components/data-management/ProjectDetails.jsx

```diff
@@ -158,6 +166,8 @@ const ProjectDetails = ({ width, height }) => {
         value={text}
         onAfterSubmit={(name) => dispatch(updateSample(el.uuid, { name }))}
         onDelete={() => dispatch(deleteSamples(el.uuid))}
+        validationFunc={(name) => validateSampleName(name, sampleNames)}
+        errorFunc={(name) => validateSampleName(name, sampleNames)}
```

Comment:
do we need to differentiate between `validationFunc` and `errorFunc`? I see that in practise, they are both equal to the same thing

Reply:
In this case, the validation function returns the error message, that's why it's set like that. I imagine there are cases where the validation function is quite simple `((name) => name !== 'projects')`, and does not return the text, then the `errorText` can be used

Reply:
are we going to have a case like this any time soon?

Reply:
Hmm.. not really. I guess we can force implementors to return error string from `validationFunc`. This would enforce the function to provide an error message during validation handling too.